packages/arb-token-bridge-ui/src/components/TransferPanel/AdvancedSettings.tsx

```diff
@@ -1,11 +1,63 @@
 import { useEffect, useState } from 'react'
+import { isAddress } from 'ethers/lib/utils'
+import { Provider } from '@ethersproject/providers'
 import { ChevronDownIcon, ChevronUpIcon } from '@heroicons/react/24/outline'
 
 import { useAppState } from '../../state'
 import { useAccountType } from '../../hooks/useAccountType'
+import { addressIsSmartContract } from '../../util/AddressUtils'
+import { useNetworksAndSigners } from '../../hooks/useNetworksAndSigners'
 
-export enum AdvancedSettingsErrors {
-  INVALID_ADDRESS = 'The destination address is not valid.'
+export enum DestinationAddressErrors {
+  INVALID_ADDRESS = 'The destination address is not a valid address.',
+  REQUIRED_ADDRESS = 'The destination address is required.'
+}
+
+enum DestinationAddressWarnings {
+  CONTRACT_ADDRESS = 'The destination address is a contract address. Please make sure it is a valid wallet address.'
+}
+
+export function getDestinationAddressError({
+  destinationAddress,
+  isSmartContractWallet
+}: {
+  destinationAddress?: string
+  isSmartContractWallet: boolean
+}): DestinationAddressErrors | null {
+  if (!destinationAddress && isSmartContractWallet) {
+    // destination address required for contract wallets
+    return DestinationAddressErrors.REQUIRED_ADDRESS
+  }
+
+  if (destinationAddress && !isAddress(destinationAddress)) {
+    return DestinationAddressErrors.INVALID_ADDRESS
+  }
+
+  // no error
+  return null
+}
+
+async function getDestinationAddressWarning({
```

Comment:
I think there was a check in here for valid address, should we add it back?

Reply:
We shouldn't need it because error will handle invalid addresses, but I will add it at the top so that we don't call async methods needlessly.